pyopencl/__init__.py

```diff
@@ -1648,7 +1648,7 @@ def parse_device(choice):
 
     if not devices:
         raise Error("no devices found")
-    elif len(devices) == 1:
+    elif len(devices) == 1 and not answers:
```

Comment:
Out of curiosity, why does the device selection have a different behavior from platform selection here, when there is only 1 "item" available - the user does not get to even see which device was chosen automatically in the 1-device case? Perhaps we could just remove this `elif` alltogether?

Reply:
Another option could be to just print the automatically chosen device in the `len == 1` case.

Reply:
The idea of the `elif` was to keep the interactive process as efficient as possible.

Reply:
> Another option could be to just print the automatically chosen device in the `len == 1` case.

I like that. Want to add it to this PR?

Reply:
> > Another option could be to just print the automatically chosen device in the `len == 1` case.
> 
> I like that. Want to add it to this PR?

Sure, what do you think of [55f3e0c](https://github.com/inducer/pyopencl/pull/780/commits/55f3e0c2a65491275a55ea25c9d69da22f179271)?

Reply:
LGTM.